jina/serve/networking.py

```diff
@@ -577,6 +585,7 @@ def add_connection(
             if shard_id is None:
                 shard_id = 0
             self._connections.add_replica(deployment, shard_id, address)
+        self._deployment_address_map[deployment] = address
```

Comment:
This mechanism is not quite correct as a single deployment can have multiple addresses (when replicas > 1 and not in K8s). I am not sure it can be easily changed for your case, but we somehow need a dict from a specific replica to a deployment and tell the user which replica with which address actually failed? (Might also make a nice test case)

Reply:
By virtue of our retry mechanism, a request will not fail if just one replica is down. Consequently, it is not possible to report this back to the user in the same way as in a non-replicated deployment, unless we break a working Flow just to do so.

So here is the design decision I would make:
- If at least one replica is still up, the request will go through and the client will not be alerted, he just gets his response. On the gateway side, we log the failed attempt, including the address of the specific replica that failed.
- If all replicas are down, the request fails in the same way as a non-replicated executor fails. In this case, the user will see all addresses of all replicas.

Does this makes sense @JoanFM ?

Reply:
What u describe makes sense, I will review if the PR is still valid, but what you describe makes a lot of sense